dashboard/src/actions/overviewActions.js

```diff
@@ -407,3 +407,137 @@ const clearEditableFields = (item) => {
   item[CONSTANTS.IS_DIRTY_NAME] = false;
   item[CONSTANTS.IS_DIRTY_SERVER_DELETE] = false;
 };
+export const setMetadataModal = (isOpen, run) => async (dispatch, getState) => {
+  dispatch({
+    type: TYPES.SET_METADATA_MODAL,
+    payload: isOpen,
+  });
+};
+/**
+ * Function to get keySummary and send to parse data
+ * @function
+ * @return {Function} - dispatch the action and update the state
+ */
+export const getKeySummary = () => async (dispatch, getState) => {
+  try {
+    const endpoints = getState().apiEndpoint.endpoints;
+    const response = await API.get(uriTemplate(endpoints, "datasets_list"), {
+      params: { keysummary: true },
+    });
+    if (response.status === 200) {
+      if (response.data.keys) {
+        dispatch({
+          type: TYPES.GET_KEY_SUMMARY,
+          payload: response.data,
+        });
+        dispatch(parseKeySummaryforTree());
+      }
+    }
+  } catch (error) {
+    dispatch(showToast(DANGER, ERROR_MSG));
+  }
+};
+
+export const parseKeySummaryforTree = () => (dispatch, getState) => {
+  const parsedData = [];
+  const keySummary = getState().overview.keySummary.keys;
+  const optionRoles = Object.keys(keySummary);
+  const checkedItems = getState().overview.checkedItems;
+
+  for (const item of optionRoles) {
+    const dataObj = { title: item, options: [] };
+    for (const key in keySummary[item]) {
+      if (!isNonEssentialKey(key)) {
+        const aggregateKey = `${item}${CONSTANTS.KEYS_JOIN_BY}${key}`;
+        const isChecked = findItemCheked(aggregateKey, checkedItems);
+        const obj = constructTreeObj(aggregateKey, isChecked);
+        if (!keySummary[item][key]) {
+          dataObj.options.push(obj);
+        } /* has children */ else {
+          obj["children"] = constructChildTreeObj(
+            aggregateKey,
+            keySummary[item][key],
+            checkedItems,
+            dispatch
+          );
+          dataObj.options.push(obj);
+        }
+      }
+    }
```

Comment:
How is this code different from `constructChildTreeObj()`?

I think the differences are around updating the checked items list (which I'm hoping you will defer until the end of the parse) and the construction of the "aggregate key" (which I think we can handle with a local conditional); if we can address those differences, then I don't think we need to duplicate this code.